File: src/HoangTLM.CodeBase.DotNetScanner/Models/CodeEntity.cs
using System;

namespace HoangTLM.CodeBase.DotNetScanner.Models
{
    public class CodeEntity
    {
        public string Id { get; set; }
        public string FileId { get; set; }
        public string Name { get; set; }
        public string Type { get; set; }
        public string Signature { get; set; }
        public int StartLine { get; set; }
        public int EndLine { get; set; }
        public string Metadata { get; set; }
        public string Description { get; set; }
        public string RelativePath { get; set; }
        public string AbsolutePath { get; set; }
    }
}

File: src/HoangTLM.CodeBase.WebScanner/src/models/code-entity.ts
export interface CodeEntity {
  id: string;
  fileId: string;
  name: string;
  type: string;
  signature: string;
  startLine: number;
  endLine: number;
  metadata: string; // JSON stringified
  description?: string;
  relativePath: string;
  absolutePath: string;
}

export interface RelationEntity {
  id: string;
  sourceId: string;
  targetId: string;
  type: string;
  metadata: string; // JSON stringified
}
